src/foam/nanos/pm/FoldReducePMLogger.java

```diff
@@ -45,23 +45,30 @@ public void fold(Object state, Object op) {
       pi = new PMInfo();
       pi.setKey(pm.getKey());
       pi.setName(pm.getName());
+      pi.setCapture(captureEnabledList_.contains(key));
```

Comment:
HashSet is not thread safe but you're accessing it from both the global FoldReducer and from the thread-local threads. InitialState() is called under synchronization (by containment from resetState()), so it would be safe to pre-populate the HashMap() there with information from captureEnabledList_.